managed-ledger/src/test/java/org/apache/bookkeeper/mledger/impl/ManagedCursorTest.java

```diff
@@ -3526,5 +3528,26 @@ public void testCursorCheckReadPositionChanged() throws Exception {
         ledger.close();
     }
 
+
+    @Test // (timeOut = 20000)
+    public void testCursorGetBacklog() throws Exception {
+        ManagedLedgerConfig managedLedgerConfig = new ManagedLedgerConfig();
+        managedLedgerConfig.setMaxEntriesPerLedger(1);
+        managedLedgerConfig.setMinimumRolloverTime(0, TimeUnit.MILLISECONDS);
+        ManagedLedgerImpl ledger = (ManagedLedgerImpl) factory.open("get-backlog", managedLedgerConfig);
+        ManagedCursor managedCursor = ledger.openCursor("test");
+
+        Position position = ledger.addEntry("test".getBytes(Encoding));
+        ledger.addEntry("test".getBytes(Encoding));
+
+        Assert.assertEquals(managedCursor.getNumberOfEntriesInBacklog(true), 2);
+        Field field = ManagedLedgerImpl.class.getDeclaredField("ledgers");
+        field.setAccessible(true);
+
+        ((ConcurrentSkipListMap<Long, MLDataFormats.ManagedLedgerInfo.LedgerInfo>) field.get(ledger)).remove(position.getLedgerId());
```

Comment:
Maybe it's better to change the `markDeletePosition` at the same time, due to the range start position is `markDeletePosition`.

Reply:
test when the ledger have been deleted and the markDeletePosition is still the deleted ledger we can get the correct backlog.